components/canvas/raqote_backend.rs

```diff
@@ -630,22 +604,47 @@ impl Filter {
     }
 }
 
-pub(crate) struct PathBuilder(Option<raqote::PathBuilder>);
+pub(crate) struct Path(Cell<raqote::PathBuilder>);
```

Comment:
Since this is accessed mutability usually, is `Cell` even necessary here? Could you just make this:

```suggestion
pub(crate) struct Path(raqote::PathBuilder);
```

Reply:
Yes, for `finish(self)`, because we need to "take" it.

Reply:
I don't see a `finish(self)` implemented anywhere. Do you mind linking to it?

Reply:
Ah, raqote::PathBuilder has finish and we use it here: https://github.com/sagudev/servo/blob/1fbd9d6657b1b06b9b3817ccacc6d533eb179202/components/canvas/raqote_backend.rs#L624

Originally this was done by wrapping raqote::PathBuilder in Option (so we can take it), but that required doing a lot of unwraps. Using cell avoids this, because there is always a pathbuilder available to manipulate.